engine/access/rest/handler.go

```diff
@@ -121,33 +156,10 @@ func (h *Handler) errorResponse(
 	responseMessage string,
 	logger zerolog.Logger,
 ) {
-	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
-	w.WriteHeader(returnCode)
-
 	// create error response model
-	modelError := generated.ModelError{
+	modelError := models.ModelError{
 		Code:    int32(returnCode),
 		Message: responseMessage,
 	}
-	encodedError, err := json.Marshal(modelError)
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		logger.Error().Str("response_message", responseMessage).Msg("failed to json encode error message")
-		return
-	}
-
-	_, err = w.Write(encodedError)
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		logger.Error().Err(err).Msg("failed to send error response")
-	}
-}
-
-// NotImplemented handler returns an error explaining the endpoint is not yet implemented
-func NotImplemented(
-	_ *request,
-	_ access.API,
-	_ LinkGenerator,
-) (interface{}, StatusError) {
-	return nil, NewRestError(http.StatusNotImplemented, "endpoint not implemented", nil)
+	h.jsonResponse(w, returnCode, modelError, logger)
```

Comment:
Yeah I was thinking about that, and technically it's possible but I can't find a plausible scenario. The scenario where this would fail is if `json.MarshalIndent` would fail to marshal `models.ModelError` which I don't see it happening, or if `_, err = w.Write(encodedResponse)` would not be successful writing that model error response. Now the second one is more complex so might happen due to weird reasons, maybe a good question is should we even try to handle that error by writing again to the response stream, or just log it and end the request. Maybe the latter could be better. The first scenario is IMHO unreachable tho, the marshaller should always work fine on this defined data. Or you think it can break?

Reply:
I'm not sure. intuition tells me if there's an error writing the data to the response stream, it would likely still be there if you retried immediately. Maybe the thing to do in the case of an error in `w.Write` is just log the error, then return a 500 with no message?

Reply:
Yeah, I kinda agree. It's a messy situation anyway.

Reply:
hey @peterargue @sideninja - created this [PR](https://github.com/onflow/flow-go/pull/1887) to address this issue and remove the recursive call. Please can you review? thanks

Reply:
merged the [PR](https://github.com/onflow/flow-go/pull/1887). @peterargue  @sideninja  - marking this comment as resolved.